src/components/ChatComponents/ChatIcons.tsx

```diff
@@ -62,31 +61,31 @@ const ChatIcons: React.FC<ChatIconsProps> = ({
     const selectedModel = event.target.value;
     setCurrentModel(event.target.value);
 
-    if (selectedModel === ChatModelDisplayNames.CLAUDE) {
-      // Start the proxy server when CLAUDE is selected
-      await proxyServer.startProxyServer('https://api.anthropic.com/');
+    // Start proxy server based on the selected model & settings
+    const proxyServerURL = proxyServer.getProxyURL(selectedModel);
+    if (proxyServerURL) {
+      await proxyServer.startProxyServer(proxyServerURL, selectedModel !== ChatModelDisplayNames.CLAUDE);
     } else {
-      // Stop the proxy server when another model is selected
       await proxyServer.stopProxyServer();
     }
   };
 
   useEffect(() => {
-    // If Claude is the default, start the proxy server
-    const startProxyServerForClaude = async () => {
-      if (currentModel === ChatModelDisplayNames.CLAUDE) {
-        await proxyServer.startProxyServer('https://api.anthropic.com/');
-      }
+    const startProxyServerForClaude = async (proxyServerURL: string) => {
+      await proxyServer.startProxyServer(proxyServerURL, currentModel !== ChatModelDisplayNames.CLAUDE);
     };
 
     // Call the function on component mount
-    startProxyServerForClaude();
+    const proxyServerURL = proxyServer.getProxyURL(currentModel);
+    if (proxyServerURL) {
+      startProxyServerForClaude(proxyServerURL);
+    }
 
     // Cleanup function to stop the proxy server when the component unmounts
     return () => {
       proxyServer.stopProxyServer().catch(console.error);
     };
-  }, [currentModel, proxyServer]);
+  }, []);
```

Comment:
Previously, including these deps caused the cleanup function to fire each time a new model was selected.